File: src/main/java/net/solvetheriddle/processing/sketches/Screen.java
package net.solvetheriddle.processing.sketches;

import processing.core.PApplet;
import processing.core.PImage;
import util.ImageLoad;

public class Screen extends PApplet {

    private static int ROWS = 12, COLS = 32, SCALE = 50;
    private static int H = ROWS * SCALE, W = COLS * SCALE;
    private static PImage img;

    public static void main(String... args) {
        PApplet.main(Screen.class.getName());
    }

    public void settings() {
        size(W, H);
        img = ImageLoad.getImage("sw.jpg");
        img.resize(32, 12);
        img.loadPixels();
        noLoop();
    }

    public void draw() {
        background(0);
        for (int row = 0; row < ROWS; row++) {
            for (int col = 0; col < COLS; col++) {
                int clr = img.get(col, row);
                fill(clr);
                rect(col * SCALE, row * SCALE, (col + 1) * SCALE, (row + 1) * SCALE);
            }
        }
    }

}

File: javaApp/src/main/java/net/solvetheriddle/processing/java/library/Screen.kt
package net.solvetheriddle.processing.java.library

import net.solvetheriddle.processing.java.utils.ImageLoad
import processing.core.PApplet
import processing.core.PImage

fun main() {
    PApplet.main(Screen::class.java.name)
}

class Screen: PApplet() {

    private val ROWS = 12
    private val COLS = 32
    private val SCALE = 50
    private val H = ROWS * SCALE
    private val W = COLS * SCALE
    private var img: PImage? = null

    override fun settings() {
        size(W, H)
        img = ImageLoad().getImage("sw.jpg")
        img!!.resize(32, 12)
        img!!.loadPixels()
        noLoop()
    }

    override fun draw() {
        background(0)
        for (row in 0 until ROWS) {
            for (col in 0 until COLS) {
                val clr = img!!.get(col, row)
                fill(clr)
                rect(
                    (col * SCALE).toFloat(),
                    (row * SCALE).toFloat(),
                    ((col + 1) * SCALE).toFloat(),
                    ((row + 1) * SCALE).toFloat()
                )
            }
        }
    }

}